qa/pull-tester/rpc-tests.py

```diff
@@ -266,7 +268,25 @@ def main():
         subprocess.check_call((config["environment"]["SRCDIR"] + '/qa/rpc-tests/' + test_list[0]).split() + ['-h'])
         sys.exit(0)
 
-    run_tests(test_list, config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
+
+    if (args.rpcgroup == -1) != (args.machines == -1):
+        print("ERROR: Please use both -m and -r options when using parallel rpc_groups.")
+        sys.exit(0)
+    if args.machines == 0:
+        print("ERROR: -m/--machines must be greater than 0 and -r/--rpcgroup must be less than -m/--machines")
+        sys.exit(0)
+    if args.rpcgroup > args.machines:
```

Comment:
```suggestion
    if args.rpcgroup >= args.machines:
```

Reply:
IIRC this breaks legacy calls from `full_test_suite.py`

Reply:
Fixed now.

Reply:
Because when there are no arguments, `args.rpcgroup == args.machines == -1` 🤦 Okay, this should fix that:

```suggestion
    if args.machines > 0 and args.rpcgroup >= args.machines:
```